drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -373,12 +395,15 @@ def _get_user_state(self):
         """ Get all user-specific data, and any applicable feedback """
         item_state = self._get_item_state()
         for item_id, item in item_state.iteritems():
-            definition = self._get_item_definition(int(item_id))
             # If information about zone is missing
             # (because problem was completed before a11y enhancements were implemented),
             # deduce zone in which item is placed from definition:
             if item.get('zone') is None:
-                item['zone'] = definition.get('zone', 'unknown')
+                valid_zones = self._get_valid_item_zones(int(item_id))
+                if valid_zones:
+                    item['zone'] = valid_zones[0]
```

Comment:
@haikuginger It might be worth mentioning that support for multiple target zones was implemented *after* the a11y enhancements mentioned above, and that this means that `valid_zones` will at most contain a single zone for items from problems that were "completed before a11y enhancements were implemented". This is not immediately obvious from looking at the code alone (if it were possible for items without `zone` info to be associated with multiple target zones, using `valid_zones[0]` would not necessarily return the zone that a student actually placed an item in).